spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/ReactiveDatastoreTemplate.java

```diff
@@ -0,0 +1,205 @@
+		Set<Key> keys = this.datastoreTemplate.getKeysFromIds(ids, entityClass);
+		return findAllById(keys, entityClass, new DatastoreTemplate.ReadContext());
+	}
+
+	private <T> Flux<T> findAllById(Set<Key> keys, Class<T> entityClass, DatastoreTemplate.ReadContext context) {
+		if (keys == null || keys.isEmpty()) {
+			return Flux.empty();
+		}
+
+		/*
+		 * We mark each key that will be read as DONE in the read context with a null value. The
+		 * null is only a placeholder to prevent circular-dependencies for when the actual entity
+		 * is read by the Flux below.
+		 */
+		for (BaseKey mk : keys) {
+			context.putReadEntity(mk, null);
+		}
+
+		return convertAndResolveRelationshipsEagerly(
+				Flux.fromIterable(
+						() -> this.datastoreTemplate.getDatastoreReadWriter().get(keys.toArray(new Key[] {}))),
+				entityClass, context);
+	}
+
+	public <T> Flux<T> findAll(Class<T> entityClass, DatastoreQueryOptions queryOptions) {
+		DatastorePersistentEntity<?> persistentEntity = this.datastoreTemplate.getDatastoreMappingContext()
+				.getPersistentEntity(entityClass);
+		EntityQuery.Builder builder = Query.newEntityQueryBuilder()
+				.setKind(persistentEntity.kindName());
+		DatastoreTemplate.applyQueryOptions(builder, queryOptions, persistentEntity);
+		Query query = builder.build();
+
+		return convertAndResolveRelationshipsEagerly(
+				Flux.fromIterable(() -> this.datastoreTemplate.getDatastoreReadWriter().run(query)), entityClass,
+				new DatastoreTemplate.ReadContext());
+	}
+
+	public <T> Flux<T> query(Query<? extends BaseEntity> query, Class<T> entityClass) {
+		return convertAndResolveRelationshipsEagerly(
+				(Flux<Entity>) Flux.fromIterable(() -> this.datastoreTemplate.getDatastoreReadWriter().run(query)),
+				entityClass, new DatastoreTemplate.ReadContext());
+	}
+
+	public <A, T> Flux<T> query(Query<A> query, Function<A, T> entityFunc) {
+		return Flux.fromIterable(() -> this.datastoreTemplate.getDatastoreReadWriter().run(query))
+				.map(entityFunc);
+	}
+
+	public Flux<Key> queryKeys(Query<Key> query) {
+		return Flux.fromIterable(() -> this.datastoreTemplate.getDatastoreReadWriter().run(query));
+	}
+
+	public <T> Flux<T> findAll(Class<T> entityClass) {
+		return findAll(entityClass, null);
+	}
+
+	public <T> Flux<T> queryByExample(Example<T> example, DatastoreQueryOptions queryOptions) {
+		return query(this.datastoreTemplate.exampleToQuery(example, queryOptions, false), example.getProbeType());
+	}
+
+	public <T> Flux<Key> keyQueryByExample(Example<T> example, DatastoreQueryOptions queryOptions) {
+		Query query = this.datastoreTemplate.exampleToQuery(example, queryOptions, true);
+		return Flux.fromIterable(() -> this.datastoreTemplate.getDatastoreReadWriter().run(query));
+	}
+
+	public <T> Mono<Boolean> existsById(Object id, Class<T> entityClass) {
+		return findById(id, entityClass).map(Objects::nonNull);
+	}
+
+	private Flux<Key> findAllKeys(Class entityClass) {
```

Comment:
Since there is no usecase where we use keys one at a time we could remove tyhis and just use the old blocking function.